packages/rpc-graphql/src/resolvers/resolve-info/account.ts

```diff
@@ -0,0 +1,95 @@
+
+function findArgumentNodeByName(argumentNodes: readonly ArgumentNode[], name: string): ArgumentNode | undefined {
+    return argumentNodes.find(argumentNode => argumentNode.name.value === name);
+}
+
+function parseAccountEncodingArgument(
+    argumentNodes: readonly ArgumentNode[],
+    variableValues: {
+        [variable: string]: unknown;
+    },
+): AccountLoaderArgs['encoding'] | undefined {
+    const argumentNode = findArgumentNodeByName(argumentNodes, 'encoding');
+    if (argumentNode) {
+        if (argumentNode.value.kind === 'EnumValue') {
+            if (argumentNode.value.value === 'BASE_58') {
+                return 'base58';
+            }
+            if (argumentNode.value.value === 'BASE_64') {
+                return 'base64';
+            }
+            if (argumentNode.value.value === 'BASE_64_ZSTD') {
+                return 'base64+zstd';
+            }
+        }
+        if (argumentNode.value.kind === 'Variable') {
+            return variableValues[argumentNode.value.name.value] as AccountLoaderArgs['encoding'];
+        }
+    } else {
+        return undefined;
+    }
+}
+
+function parseAccountDataSliceArgument(
+    argumentNodes: readonly ArgumentNode[],
+    variableValues: { [variable: string]: unknown },
+): AccountLoaderArgs['dataSlice'] {
+    const argumentNode = findArgumentNodeByName(argumentNodes, 'dataSlice');
+    if (argumentNode) {
+        if (argumentNode.value.kind === 'ObjectValue') {
+            const offsetArg = argumentNode.value.fields?.find(field => field.name.value === 'offset');
+            const lengthArg = argumentNode.value.fields?.find(field => field.name.value === 'length');
+            const length = lengthArg?.value.kind === 'IntValue' ? parseInt(lengthArg.value.value) : undefined;
+            const offset = offsetArg?.value.kind === 'IntValue' ? parseInt(offsetArg.value.value) : undefined;
+            return length !== undefined && length !== null && offset !== undefined && offset !== null
+                ? { length, offset }
+                : undefined;
+        }
+        if (argumentNode.value.kind === 'Variable') {
+            return variableValues[argumentNode.value.name.value] as DataSlice;
+        }
+    } else {
+        return undefined;
+    }
+}
+
+/**
+ * Build a set of account loader args by inspecting which fields have
+ * been requested in the query (ie. `data` or inline fragments).
+ */
+export function buildAccountLoaderArgSetFromResolveInfo(
+    args: {
+        address: Address;
+        commitment?: Commitment;
+        minContextSlot?: Slot;
+    },
+    info?: GraphQLResolveInfo,
+): AccountLoaderArgs[] {
+    const argSet: AccountLoaderArgs[] = [args];
+
+    if (info?.fieldNodes[0]?.selectionSet) {
+        const { variableValues } = info;
+        const selections = info.fieldNodes[0].selectionSet.selections;
+
+        if (isInlineFragmentPresent(selections)) {
```

Comment:
Isn't this _technically_ insufficient to determine if the account should be fetched with `jsonParsed` or not? Couldn't there be some kind of type in this union that doesn't actually require that we fetch parsed data? It would be more airtight to actually check the fragment's `typeCondition` against a known list of types.

¯\\_(°ペ)\_/¯

Reply:
aaaand done!